### PR TITLE
Disputable voting app: State reducer for vote information

### PR DESCRIPTION
**Pull request by delfipolito**

### CODE REVIEW

File: apps/voting-disputable/app/src/script.js [hunk 1]
```diff
@@ -358,6 +359,75 @@ function loadVoteDisputableInfo(voteId) {
...
+async function getDisputableAction(actionId) {
+  const agreementAddress = await getAgreement()
+  const agreement = app.external(agreementAddress, agreementAbi)
+
+  return agreement
+    .getAction(actionId)
+    .toPromise()
+    .then(action => getChallenge(agreement, action))
+    .then(action => getCollaterall(agreement, action))
+    .then(action => getCollaterallToken(action))
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/script.js:383]: Can we perform these concurrently via `promise.all`?

File: apps/voting-disputable/app/src/script.js [hunk 2]
```diff
@@ -358,6 +359,75 @@ function loadVoteDisputableInfo(voteId) {
...
+      ...action,
+      challenge: await agreement
+        .getChallenge(action.currentChallengeId)
+        .toPromise(),
+    }
+  }
+  return action
+}
+
+async function getCollaterall(agreement, action) {
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/script.js:398]: I think these function names should probably be updated as `Collaterall` -> `Collateral`

File: apps/voting-disputable/app/src/script.js [hunk 3]
```diff
@@ -426,12 +496,20 @@ function marshallVote({
...
+  status,
+}) {
   return {
     actionId,
     pauseDuration,
     pausedAt: marshallDate(pausedAt),
     status,
+    action: {
+      ...(await getDisputableAction(actionId)),
+    },
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/script.js:512]: do we need to spread here? or can this be assigned simply as:

```javascript
action: await getDisputableAction(actionId),
```

File: apps/voting-disputable/app/src/script.js [hunk 4]
```diff
@@ -358,6 +359,75 @@ function loadVoteDisputableInfo(voteId) {
...
+    action.collateral.collateralToken,
+    tokenAbi
+  )
+  if (action.collateral && action.collateral.collateralToken) {
+    return {
+      ...action,
+      collateral: {
+        ...action.collateral,
+        symbol: await collateralToken.symbol().toPromise(),
+        decimals: await collateralToken.decimals().toPromise(),
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/script.js:424]: Could these also be performed concurrently?

File: apps/voting-disputable/app/src/components/VoteDetail/DisputableActionStatus.js [hunk 5]
```diff
@@ -21,7 +30,7 @@ function DisputableActionStatus({ vote }) {
   //TODO: get agreement and vote real data
   const agreement = getAgreement()
 
-  const { challengeAmount, collateralToken } = vote.disputable.action.collateral
+  const { actionAmount, decimals, symbol } = vote.disputable.action.collateral
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/components/VoteDetail/DisputableActionStatus.js:33]: I think this is going to throw as `disputable` is initially `undefined`, I also don't think we need to provide the entire vote object to this component. How about just passing `disputable` to props and then guarding appropriately? e.g.

```javascript
const tokenAmount = useMemo(() => {
    if (disputable) {
      const { actionAmount, decimals, symbol } = disputable.action.collateral

      return formatTokenAmount(actionAmount, decimals, { symbol: symbol })
    }

    return ''
  }, [disputable])

```

File: apps/voting-disputable/app/src/components/VoteDetail/DisputableActionStatus.js [hunk 6]
```diff
@@ -21,7 +30,7 @@ function DisputableActionStatus({ vote }) {
   //TODO: get agreement and vote real data
   const agreement = getAgreement()
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/components/VoteDetail/DisputableActionStatus.js:31]: Just checking, do we still need this?

File: apps/voting-disputable/app/src/script.js [hunk 7]
```diff
@@ -347,7 +348,7 @@ function loadVoteDisputableInfo(voteId) {
     app
       .call('getVoteDisputableInfo', voteId)
       .toPromise()
-      .then(vote => marshallDisputableInfo(vote))
+      .then(disputable => marshallDisputableInfo(disputable))
```
Comment (andy-hook) [inline apps/voting-disputable/app/src/script.js:351]: Do you think passing as `disputableInfo` is slightly clearer?

**Comment by delfipolito** [comment]
@andy-hook I will merge this to allow testing on the client but I will address your comments when I move this code to the new repo :)

**Pull request merged**